From 3ec5e5a14ef4e6762667b1d6f2eb7f1159063197 Mon Sep 17 00:00:00 2001
From: Rajab Natshah <33889-RajabNatshah@users.noreply.drupalcode.org>
Date: Mon, 1 Jul 2024 14:02:01 +0000
Subject: [PATCH] Issue #3457431: Fix validation warnings when saving text
 format by remove inactive filters from the list of filters to validate for
 the text format

---
 core/modules/ckeditor5/src/Plugin/Editor/CKEditor5.php | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/core/modules/ckeditor5/src/Plugin/Editor/CKEditor5.php b/core/modules/ckeditor5/src/Plugin/Editor/CKEditor5.php
index 821f4cc3b1b9..f7475dae2cd2 100644
--- a/core/modules/ckeditor5/src/Plugin/Editor/CKEditor5.php
+++ b/core/modules/ckeditor5/src/Plugin/Editor/CKEditor5.php
@@ -212,6 +212,14 @@ public static function validatePair(EditorInterface $text_editor, FilterFormatIn
       throw new \InvalidArgumentException('This text editor is not configured to use CKEditor 5.');
     }
 
+    // Remove inactive filters from the list of filters to validate for the text format.
+    $text_format_filters = [];
+    foreach ($text_format->toArray()['filters'] as $filter_index => $filter) {
+      if ($filter['status']) {
+        $text_format_filters[$filter_index] = $filter;
+      }
+    }
+
     $typed_config_manager = \Drupal::getContainer()->get('config.typed');
     $typed_config = $typed_config_manager->createFromNameAndData(
       'ckeditor5_valid_pair__format_and_editor',
@@ -224,7 +232,7 @@ public static function validatePair(EditorInterface $text_editor, FilterFormatIn
         // - filter.format.*.filters — note that "filters" is top-level in
         //   filter.format.*, and so it is here, so all validation constraints
         //   will continue to work fine.
-        'filters' => $text_format->toArray()['filters'],
+        'filters' => $text_format_filters,
         // - editor.editor.*.image_upload — note that "image_upload" is
         //   top-level in editor.editor.*, and so it is here, so all validation
         //   constraints will continue to work fine.
-- 
GitLab

